src/main/java/org/openrewrite/java/spring/boot2/LoggingShutdownHooksRecipe.java

```diff
@@ -0,0 +1,76 @@
+package org.openrewrite.java.spring.boot2;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.MavenVisitor;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.maven.tree.Pom;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Collection;
+
+/**
+ * @author Alex Boyko
+ */
+public class LoggingShutdownHooksRecipe extends Recipe {
+
+    private static final XPathMatcher PROJECT_MATCHER = new XPathMatcher("/project");
+
+    private static final String SEARCH_MARKER_DESCRIPTION = "LoggingShutDownHook";
+
+    @Override
+    public String getDisplayName() {
+        return "Logging Shutdown Hooks";
+    }
+
+    @Override
+    public String getDescription() {
+        return "We now register a logging shutdown hook by default for jar based applications to ensure that " +
+                "logging resources are released when the JVM exits. If your application is deployed as a war then " +
+                "the shutdown hook is not registered since the servlet container usually handles logging concerns" +
+                "\n\n" +
+                "Most applications will want the shutdown hook. However, if your application has complex context" +
+                " hierarchies, then you may need to disable it. You can use the logging.register-shutdown-hook" +
+                " property to do that.";
+    }
+
+    @Override
+    protected @Nullable TreeVisitor<?, ExecutionContext> getApplicableTest() {
```

Comment:
a suggestion for the applicable test would be to do something similar to the DependencyInsight recipe which will find both direct and transitive dependencies. Also moving the packaging check to the recipe visitor will improve recipe performace by not having to inspect every tag of every pom 
```java
@Override
protected MavenVisitor getApplicableTest() {
    return new MavenVisitor() {
        @Override
        public Xml visitTag(Xml.Tag tag, ExecutionContext context) {
            Xml.Tag t = (Xml.Tag) super.visitTag(tag, context);
            if (isDependencyTag()) {
                Pom.Dependency dependency = findDependency(t);
                if (dependency != null) {
                    Collection<Pom.Dependency> match = dependency.findDependencies("org.springframework.boot", "spring-boot");
                    if (match.stream().anyMatch(dep -> dep.getVersion().startsWith("2.5"))) {
                        t = t.withMarkers(t.getMarkers().searchResult());
                    }
                }
            }
            return t;
        }
    };
}
```